app/routes/user/password.tsx

```diff
@@ -0,0 +1,276 @@
+  const validLength = newPassword.length >= 8
+  const containsSpecialChar = /[~`!@#$%^.&*+=\-_ [\]\\';,/{}()|\\":<>?]/g.test(
+    newPassword
+  )
+  const leadingOrTrailingSpace =
+    newPassword.length !== newPassword.trim().length
+  const oldPasswordError = errorMessage && !isOldPasswordTouched
+  const valid = !checkWhitespace(newPassword) && checkPassword(newPassword)
+  const shouldDisableSubmit =
+    !newPassword || !isNewPasswordTouched || !valid || !isOldPasswordTouched
+  let isError
+  if (isNewPasswordTouched && valid) {
+    isError = false
+  }
+  if (isNewPasswordTouched && !valid) {
+    isError = true
+  }
+  const formRef = useRef<HTMLFormElement>(null)
+
+  useEffect(() => {
+    if (formRef.current && fetcher.state === 'submitting') {
+      formRef.current.reset()
+    }
+  }, [fetcher.state])
+
+  return (
+    <>
+      <h1>Reset Password</h1>
+      <>
+        <fetcher.Form method="POST" ref={formRef}>
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <TextInput
+            data-focus
+            className={oldPasswordError ? 'usa-input--error' : ''}
+            name="oldPassword"
+            id="oldPassword"
+            type="password"
+            placeholder="Old Password"
+            onChange={(e) => {
+              setIsOldPasswordTouched(true)
+            }}
+          />
+          {errorMessage === 'NotAuthorizedException' && (
+            <span className="text-red">Invalid Password</span>
+          )}
+          {errorMessage === 'LimitExceededException' && (
+            <span className="text-red">
+              Attempts Exceeded. Please try again later.
+            </span>
+          )}
+          <Label htmlFor="newPassword">New Password</Label>
+          <InputGroup error={isError}>
+            <TextInput
+              name="newPassword"
+              id="newPassword"
+              className={
+                !isError && isNewPasswordTouched ? 'usa-input--success' : ''
+              }
+              type={visible ? 'text' : 'password'}
+              placeholder="New Password"
+              onChange={(e) => {
+                setNewPassword(e.target.value)
+                setIsNewPasswordTouched(true)
+              }}
+              value={newPassword}
+            />
+            {visible && (
+              <div className="usa-input-suffix" aria-hidden="true">
+                <Icon.VisibilityOff
+                  className="margin-y-2"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              </div>
+            )}
+            {!visible && (
+              <div className="usa-input-suffix" aria-hidden="true">
+                <Icon.Visibility
+                  className="margin-y-2"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              </div>
+            )}
```

Comment:
- [ ] Refactor to reduce code duplication.
- [ ] Add accessibility attributes to inform screen readers what this control does.